mp/src/game/client/hl2/hud_credits.cpp

```diff
@@ -743,4 +748,23 @@ void CHudCredits::MsgFunc_LogoTimeMsg(bf_read &msg)
 {
     m_iCreditsType = CREDITS_LOGO;
     PrepareLogo(msg.ReadFloat());
-}
\ No newline at end of file
+}
+
+void CHudCredits::HideScoreboardAndLobby()
+{
+    if (gViewPortInterface)
+    {
+        IViewPortPanel *showtimesPanel = gViewPortInterface->FindPanelByName(PANEL_TIMES);
+        IViewPortPanel *showlobbyPanel = gViewPortInterface->FindPanelByName(PANEL_LOBBY_MEMBERS);
```

Comment:
Finding panel by name should not be done every frame. In fact, on line 55 in this file, there's a `g_bRollingCredits` bool that's globally accessible from anywhere in the client DLL. So, theoretically, you can go to both of these panels, and in their `SetVisible` methods, check if the credits are rolling and if so, just return without setting visible (still allow hiding the panels though).